fitbenchmarking/fitbenchmarking
# 218 implement gsl

#### Description of Work

Fixes #218 


#### Testing Instructions

1. Add gsl as a software in, e.g., example_runScripts.py
2. Check results are consistent with other minimizers

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
